### ISSUE TITLE
[BUG] Undefined property error in Laravel

### ISSUE THREAD

**Issue by kjdev** [DataDog/dd-trace-php#1518]
### Bug description

When retrieving Datadog Span in Exception Handler of Laravel Framework `Undefined property: DDTrace\SpanData::$id` error.

PHP Code.
``` php
<?php
// app/Exceptions/Handler.php

namespace App\Exceptions;

use Illuminate\Foundation\Exceptions\Handler as ExceptionHandler;
use Throwable;

class Handler extends ExceptionHandler
{
    public function register()
    {
        $this->reportable(function (Throwable $e) {
            // Get Datadog span
            $span = (\DDTrace\GlobalTracer::get())->getActiveSpan();
            // use: $span->getTraceId(), $span->getSpanId()
        });
        return false;
    }
}
```

Error Message.
```
ErrorException: Undefined property: DDTrace\SpanData::$id in file /usr/share/php8/datadog_trace/bridge/_generated_tracer.php on line 867

#0 /app/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(228): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()
#1 /usr/share/php8/datadog_trace/bridge/_generated_tracer.php(867): Illuminate\Foundation\Bootstrap\HandleExceptions->Illuminate\Foundation\Bootstrap\{closure}()
#2 /usr/share/php8/datadog_trace/bridge/_generated_tracer.php(806): DDTrace\ScopeManager->reconcileInternalAndUserland()
#3 /usr/share/php8/datadog_trace/bridge/_generated_tracer.php(1053): DDTrace\ScopeManager->getActive()
#4 /app/app/Exceptions/Handler.php(40): DDTrace\Tracer->getActiveSpan()
#5 [internal function]: App\Exceptions\Handler->App\Exceptions\{closure}()
#6 /app/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/ReportableHandler.php(45): call_user_func()
#7 /app/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(233): Illuminate\Foundation\Exceptions\ReportableHandler->__invoke()
#8 /app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(49): Illuminate\Foundation\Exceptions\Handler->report()
#9 /app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(172): Illuminate\Routing\Pipeline->handleException()
#10 /app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#11 /app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(142): Illuminate\Pipeline\Pipeline->then()
#12 /app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(111): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter()
#13 /app/public/index.php(52): Illuminate\Foundation\Http\Kernel->handle()
#14 /app/vendor/laravel/framework/src/Illuminate/Foundation/resources/server.php(16): require_once('...')
#15 {main}
```

`DDTrace\SpanData` object in `active_span()` does not have id property.
``` php
object(DDTrace\SpanData)# (2) {
  ["meta"]=>
  array(0) {
  }
  ["metrics"]=>
  array(0) {
  }
}
```

Doesn't happen if Intagration is disabled in Lavavel Framework (`DD_TRACE_LARAVEL_ENABLED=false`).

``` php
\DDTrace\hook_method('Illuminate\Routing\Pipeline', 'handleException', function ($This, $scope, $args) use($rootSpan, $integration) {
    if (empty($rootSpan->exception)) {
        $integration->setError($rootSpan, $args[1]);
    }
});
````
The error would not have occurred without the hook setting.

### PHP version

```
PHP 8.0.14 (cli) (built: Dec 18 2021 21:57:50) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.14, Copyright (c) Zend Technologies
    with ddtrace v0.70.0, Copyright Datadog, by Datadog
```

### Tracer version

```
0.70.0
```

### Installed extensions

```
[PHP Modules]
Core
curl
date
ddtrace
dom
fileinfo
filter
hash
iconv
json
libxml
mbstring
openssl
pcre
Phar
phpdbg_webhelper
readline
Reflection
session
sockets
SPL
standard
tokenizer
xml
xmlreader
xmlwriter
zip
zlib

[Zend Modules]
ddtrace
```

### OS info

```
NAME="Alpine Linux"
VERSION_ID=3.15.0
PRETTY_NAME="Alpine Linux v3.15"
```

#### Output of phpinfo() (ddtrace >= 0.47.0)

DATADOG TRACER CONFIGURATION 
```
{"date":"2022-02-17T03:11:09Z","os_name":"Linux ce3d2139d3ac 5.16.8-200.fc35.x86_64 #1 SMP PREEMPT Tue Feb 8 20:58:59 UTC 2022 x86_64","os_version":"5.16.8-200.fc35.x86_64","version":"0.70.0","lang":"php","lang_version":"8.0.14","env":null,"enabled":true,"service":null,"enabled_cli":true,"agent_url":"http:\/\/localhost:8126","debug":false,"analytics_enabled":false,"sample_rate":1,"sampling_rules":[],"tags":[],"service_mapping":[],"distributed_tracing_enabled":true,"priority_sampling_enabled":true,"dd_version":null,"architecture":"x86_64","sapi":"cli-server","datadog.trace.request_init_hook":"\/usr\/share\/php8\/datadog_trace\/bridge\/dd_wrap_autoloader.php","open_basedir_configured":false,"uri_fragment_regex":null,"uri_mapping_incoming":null,"uri_mapping_outgoing":null,"auto_flush_enabled":false,"generate_root_span":true,"http_client_split_by_domain":false,"measure_compile_time":true,"report_hostname_on_root_span":false,"traced_internal_functions":null,"auto_prepend_file_configured":false,"integrations_disabled":"default","enabled_from_env":true,"opcache.file_cache":null,"agent_error":"Failed to connect to localhost port 8126 after 0 ms: Connection refused"} 
```

**Comment by bwoebi** [comment]
Looks like we have an old hack in our code:

https://github.com/DataDog/dd-trace-php/blob/master/ext/php8/php8/engine_hooks.c#L435-L441

which now just fell again on our feet :-) This effectively has the same root cause than #1358: hooking pushes a dummy span on the span stack.

For now, I'm going to implement a quick workaround (hiding this dummy span away when getting the DDTrace\active_span(), if it's not a root span). We are in progress of refactoring the internal hooking mechanism altogether, avoiding that dummy span as well.

Thanks for reporting this!

**Comment by bwoebi** [comment]
This fix will be available in the next release (0.71.0) :-)
